src/main/java/edu/harvard/iq/dataverse/dataaccess/ImageThumbConverter.java

```diff
@@ -434,59 +428,108 @@ private static boolean isThumbnailCached(StorageIO<DataFile> storageIO, int size
      */
     public static String getImageThumbnailAsBase64(DataFile file, int size) {
 
-        logger.fine("entering getImageThumbnailAsBase64, size " + size+", for "+file.getStorageIdentifier());
-         try {
-             StorageIO<DataFile> storageIO = file.getStorageIO();
+        logger.fine("entering getImageThumbnailAsBase64, size " + size + ", for " + file.getStorageIdentifier());
+
+        // if thumbnails are not even supported on this file type, no need
+        // to check anything else:
+        if (file == null || !FileUtil.isThumbnailSupported(file)) {
+            logger.fine("No thumbnail support for " + file.getContentType());
+            return null;
+        }
+
+        StorageIO<DataFile> storageIO = null;
+
+        try {
+            storageIO = file.getStorageIO();
+        } catch (IOException ioEx) {
+            logger.fine("Caught an exception while trying to obtain a thumbnail as Base64 string - could not open StorageIO on the datafile.");
+            return null;
+        }
+
+        if (storageIO == null) {
+            return null;
+        }
+
+        // skip the "isAvailable()" check - and just try to open the cached object. 
+        // if we can't open it, then we'll try to generate it. In other words, we are doing it in 
+        // the reverse order - and his way we can save one extra lookup, for a thumbnail 
+        // that's already cached - and on some storage media (specifically, S3)
+        // lookups are actually more expensive than reads. 
```

Comment:
what's the difference between a lookup and a read?

Reply:
"check if the object exists", "retrieve the object's metadata, such as size, etc." - these are lookups. As opposed to reading the actual bytes associated with the object. 
In the case of S3 this is a lookup:
size = s3.getObjectMetadata(bucketName,destinationKey).getContentLength();

and this is a read: 

S3Object s3object = s3.getObject(new GetObjectRequest(bucketName, destinationKey));
InputStream in = s3object.getObjectContent();
in.read(...); etc. 

With the thumbnails on the dataset page - which are bunch of tiny files that we need to read to load the page - all the extra lookups - "is the main file stored?", "is the thumbnail already cached?" (we were doing this lookup more than once too, apparently), "what is the size of the cached thumbnail?" were adding up to a surprising amount of extra time. Just going directly for the bytes improved it significantly.